src/napari/_qt/qt_viewer.py

```diff
@@ -167,24 +161,19 @@ class QtViewer(QSplitter):
         A QtPoll object required for the monitor.
     _remote_manager : napari.components.experimental.remote.RemoteManager
         A remote manager processing commands from remote clients and sending out messages when polled.
-    _welcome_widget : napari._qt.widgets.qt_welcome.QtWidgetOverlay
-        QtWidgetOverlay providing the stacked widgets for the welcome page.
     """
 
     _instances = WeakSet()
 
     def __init__(
         self,
         viewer: ViewerModel,
-        show_welcome_screen: bool = False,
```

Comment:
If anyone embeds napari in their own gui, then such a person does not want to show napari welcome screen. API to disable should be keept. 

Especially this is a breaking change that will crash any app using this flag with an arguments error.